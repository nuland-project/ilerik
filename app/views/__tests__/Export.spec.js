import { render } from '@testing-library/react-native';
import React from 'react';

import { ExportScreen } from '../Export';

describe('<ExportScreen />', () => {
  it('renders correctly', () => {
    const { asJSON } = render(<ExportScreen />);
    expect(asJSON()).toMatchSnapshot();
  });

  describe('Share button', () => {
    it('should be disabled if not data is in the log', () => {
      const component = create(<Export />);
      const instance = component.root;
      const shareButton = instance.findAllByType(TouchableOpacity)[1];
      expect(shareButton.props.disabled).toBeTruthy();
    });

    it('should be disabled if not data is in the log', () => {
      const component = create(<Export shareButtonDisabled={false} />);
      const instance = component.root;
      const shareButton = instance.findAllByType(TouchableOpacity)[1];
      expect(shareButton.props.disabled).toBeFalsy();
    });
  })
});
