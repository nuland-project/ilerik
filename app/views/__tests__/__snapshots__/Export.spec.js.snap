// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportScreen /> renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(189, 195, 199,0.6)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 60,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "column",
        "padding": 20,
        "textAlignVertical": "top",
        "width": "96%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "marginTop": 12,
        }
      }
    >
      You can share your location trail with anyone using the Share button below.  Once you press the button it will ask you with whom and how you want to share it.
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "marginTop": 12,
        }
      }
    >
      Location is shared as a simple list of times and coordinates, no other identifying information.
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#665eff",
          "borderRadius": 12,
          "height": 52,
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 0.7,
          "width": 589.9499999999999,
        }
      }
    >
      <Text
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 14,
              "letterSpacing": 0,
              "lineHeight": 19,
              "textAlign": "center",
            },
            Object {
              "opacity": 0.7,
            },
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingTop": 48,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "IBMPlexSans-Medium",
                  "fontSize": 26,
                  "fontWeight": "normal",
                  "lineHeight": 24,
                  "marginTop": 9,
                  "writingDirection": "ltr",
                }
              }
              use="body1"
            >
              Share location history
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 22.5,
                  "marginTop": 22,
                  "writingDirection": "ltr",
                }
              }
              use="body1"
            >
              If you test positive for COVID-19, please do your part by sharing your location history with local authorities.
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "IBMPlexSans",
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "lineHeight": 22.5,
                  "marginTop": 22,
                  "writingDirection": "ltr",
                }
              }
              use="body1"
            >
              Location is shared as a simple list of times and places, no additional information.
            </Text>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFF",
                  "borderRadius": 8,
                  "flexDirection": "row",
                  "height": 64,
                  "justifyContent": "space-around",
                  "marginTop": 48,
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#4051DB",
                    "fontFamily": "IBMPlexSans-Medium",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Share location data
              </Text>
              <SvgXml
                style={
                  Object {
                    "height": 21,
                    "width": 16,
                  }
                }
                xml="<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<svg width=\\"16\\" height=\\"21\\" viewBox=\\"0 0 16 21\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<path d=\\"M13 7H14C15.1046 7 16 7.89543 16 9V19C16 20.1046 15.1046 21 14 21H2C0.89543 21 0 20.1046 0 19V9C0 7.89543 0.89543 7 2 7H3C3.55228 7 4 7.44772 4 8C4 8.55228 3.55228 9 3 9H2V19H14V9H13C12.4477 9 12 8.55228 12 8C12 7.44772 12.4477 7 13 7ZM7.24029 0.349711C7.42369 0.135645 7.69601 0 8 0C8.30399 0 8.57631 0.135645 8.75971 0.349711L11.5355 3.12553C11.9261 3.51606 11.9261 4.14922 11.5355 4.53975C11.145 4.93027 10.5118 4.93027 10.1213 4.53975L9 3.41843V13C9 13.5523 8.55228 14 8 14C7.44772 14 7 13.5523 7 13V3.41843L5.87868 4.53975C5.48815 4.93027 4.85499 4.93027 4.46447 4.53975C4.07394 4.14922 4.07394 3.51606 4.46447 3.12553L7.24029 0.349711Z\\" fill=\\"#4051DB\\"/>
</svg>"
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </BVLinearGradient>
  </RCTSafeAreaView>
</View>
`;
